compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ExpressionCodegen.kt

```diff
@@ -446,10 +447,10 @@ class ExpressionCodegen(
     override fun visitSetField(expression: IrSetField, data: BlockInfo): PromisedValue {
         val expressionValue = expression.value
         // Do not add redundant field initializers that initialize to default values.
-        // "expression.origin == null" means that the field is initialized when it is declared,
-        // i.e., not in an initializer block or constructor body.
-        val skip = irFunction is IrConstructor && irFunction.isPrimary &&
-                expression.origin == null && expressionValue is IrConst<*> &&
+        val inConstructor = irFunction is IrConstructor && irFunction.isPrimary
```

Comment:
inConstructor -> inPrimaryConstructor

Reply:
Please also keep  comment
//"expression.origin == null" means that the field is initialized when it is declared...

Reply:
I felt that the name of the variable made it sufficiently clear that this is an initializer rather than an assignment, but if not, then I'll leave the comment.